base/strings/util.jl

```diff
@@ -67,6 +67,25 @@ function startswith(a::Union{String, SubString{String}},
     end
 end
 
+"""
+    startswith(io::IO, prefix::Union{AbstractString,Base.Chars})
+
+Check if an `IO` object starts with a prefix.
+"""
+function Base.startswith(io::IO, prefix::Base.Chars)
+    mark(io)
```

Comment:
Seems worth it to optimize the common case of an ASCII char:
```suggestion
    prefix isa AbstractChar && prefix ≤ '\x7f' && return peek(io) == prefix % UInt8
    mark(io)
```